mantidproject/mantid
# Elemental analysis bugs

**Description of work.**
Elemental analysis had several bugs:
1. Did not work as it had Python 2 only code
2. The co-add did not work
3. The plotting did not work


**To test:**
Open elemental analysis
Tick one of the detectors
Load some data
A plot should appear
Load some different data
The plot should be replaced

Select some data and press co-add instead of load (e.g. 2695)
Change the run number and press co-add again (e.g. 2696)
You should have some co-added runs in the ADS (2695-2696)
If you tick the detector a plot should appear



No associated issue



In release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
